Included Erc-4626 in token standards- [Fixes #6255] (#6256)
ethereum/ethereum-org-website: src/content/developers/docs/standards/tokens/erc-4626/index.md

---
title: ERC-4626 Tokenized Vault Standard
description: A standard for yield bearing vaults.
lang: en
sidebar: true
---

## Introduction {#introduction}

> A standard to optimize and unify the technical parameters of yield-bearing vaults. This standard provides a standard API for tokenized yield-bearing vaults that represent shares of a single underlying ERC-20 token.
> With an optional extension for tokenized vaults utilizing ERC-20, this standard offers basic functionality for depositing, withdrawing tokens and reading balances.

Reviewer: ```suggestion
ERC-4626 is a standard to optimize and unify the technical parameters of yield-bearing vaults. It provides a standard API for tokenized yield-bearing vaults that represent shares of a single underlying ERC-20 token. ERC-4626 also outlines an optional extension for tokenized vaults utilizing ERC-20, offering basic functionality for depositing, withdrawing tokens and reading balances.
```

## ERC-4626 Functions and Features: {#body}
- [Methods](#methods)
  - [deposit](#deposit)
  - [withdraw](#withdraw)
  - [totalHoldings](#totalholdings)
  - [balanceOfUnderlying](#balanceofunderlying)
  - [underlying](#underlying)
  - [totalSupply](#totalsupply)
  - [balanceOf](#balanceof)
  - [redeem](#redeem)
  - [exchangeRate](#exchangerate)
  - [baseUnit](#baseunit)

- [Events](#events)
  - [Deposit Event](#deposit_event)
  - [Withdraw Event](#withdraw_event)

### Methods {#methods}
#### deposit {#deposit}
```solidity
function deposit(address _to, uint256 _value) public returns (uint256 _shares)
```
This function deposits the `_value` tokens into the vault and grants ownership to `_to`.

#### withdraw {#withdraw}
```solidity
function withdraw(address _to, uint256 _value) public returns (uint256 _shares)
```
This function withdraws `_value` token from the vault and transfers them to `_to`.

#### totalHoldings {#totalholdings}
```solidity
function totalHoldings() public view returns (uint256)
```
This function returns the total amount of underlying tokens held by the vault.

> #### balanceOfUnderlying {#balanceofunderlying}

Reviewer: ```suggestion
#### balanceOfUnderlying {#balanceofunderlying}

```

```solidity
function balanceOfUnderlying(address _owner) public view returns (uint256)
```
This function returns the total amount of underlying tokens held in the vault for `_owner`.

> #### underlying {#underlying}

Reviewer: ```suggestion
#### underlying {#underlying}

```

```solidity
function underlying() public view returns (address)
```
Returns the address of the token the vault uses for accounting, depositing, and withdrawing.

> #### totalSupply {#totalsupply}

Reviewer: ```suggestion
#### totalSupply {#totalsupply}

```

```solidity
function totalSupply() public view returns (uint256)
```
Returns the total number of unredeemed vault shares in circulation.

#### balanceOf {#balanceof}
```solidity
function balanceOf(address _owner) public view returns (uint256)
```
Returns the total amount of vault shares the `_owner` currently has.

> #### redeem {#redeem}

Reviewer: ```suggestion
#### redeem {#redeem}

```

```solidity
function redeem(address _to, uint256 _shares) public returns (uint256 _value)
```
Redeems a specific number of `_shares` for underlying tokens and transfers them to `_to`.

> #### exchangeRate {#exchangerate}

Reviewer: ```suggestion
#### exchangeRate {#exchangerate}

```

```solidity
function exchangeRate() public view returns (uint256)
```
The amount of underlying tokens one `baseUnit` of vault shares is redeemable for. For example:
```solidity
_shares * exchangeRate() / baseUnit() = _value.
```

```solidity
exchangeRate() * totalSupply() MUST equal totalHoldings().
```

> #### baseUnit {#baseunit}

Reviewer: ```suggestion
#### baseUnit {#baseunit}

```

```solidity
function baseUnit() public view returns(uint256)
```
The decimal scalar for vault shares and operations involving `exchangeRate()`.

> ### Events {#events}
> #### Deposit Event

Reviewer: ```suggestion
### Events {#events}

#### Deposit Event

```

**MUST** be emitted when tokens are deposited into the vault
```solidity
event Deposit(address indexed _from, addres indexed _to, uint256 _value)
```

Where `_from` is the user who triggered the deposit and approved `_value` underlying tokens to the vault, and `_to` is the user who can withdraw the deposited tokens.

> #### Widthdraw Event

Reviewer: ```suggestion
#### Widthdraw Event

```
